From 539e6a92e61ac3a6ec6ad04ad96522f0c28d6318 Mon Sep 17 00:00:00 2001
From: Vidya Sagar Velumuri <vvelumuri@marvell.com>
Date: Mon, 12 Apr 2021 05:28:52 +0000
Subject: [PATCH 1419/1921] octeontx2-af: Configure nix with proper cpt block

Update the inline configuration of nix with cpt block to which nix will
push the IPsec packets for inline processing

Signed-off-by: Vidya Sagar Velumuri <vvelumuri@marvell.com>
Change-Id: I9346ce04982d380bbba66c648e5ef6ade316aeee
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/49686
Reviewed-by: Jerin Jacob Kollanukkaran <jerinj@marvell.com>
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/af/rvu_nix.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/rvu_nix.c b/drivers/net/ethernet/marvell/octeontx2/af/rvu_nix.c
index 5396582da2ce..a97537f2b22d 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/rvu_nix.c
+++ b/drivers/net/ethernet/marvell/octeontx2/af/rvu_nix.c
@@ -4488,6 +4488,9 @@ int rvu_mbox_handler_nix_inline_ipsec_cfg(struct rvu *rvu,
 		/* Set CPT queue for inline IPSec */
 		val = (u64)req->inst_qsel.cpt_pf_func << 8 |
 		      req->inst_qsel.cpt_slot;
+
+		if (!is_rvu_otx2(rvu))
+			val |= BLKADDR_CPT0 << 24;
 		rvu_write64(rvu, blkaddr, NIX_AF_RX_CPTX_INST_QSEL(0), val);
 
 		/* Set CPT credit */
-- 
2.31.1

